torch_xla/experimental/distributed_checkpoint/manager.py

```diff
@@ -215,6 +202,35 @@ def _release_oldest_checkpoints(self):
         oldest_chkpt = self._tracked_chkpts.popleft()
         self._delete_chkpt_at_step(oldest_chkpt.step)
 
+  def _wait_for_data(self):
+    xm.mark_step()
```

Comment:
you can do `xm.mark_step(wait=True)` instead.

Reply:
I noticed that `xm.mark_step(wait=True)` doesn't actually block on device ops - I guess that's a bug. I'll open a follow-up item for this.